python-checks/src/test/resources/checks/tests/implicitlySkippedTest.py

```diff
@@ -0,0 +1,237 @@
+    if x:
+        ...
+    else:
+        ...
+    if not external_resource_available():
+        return  # Accepted FN
+    assert foo() == x
+
+
+# Unittest
+class MyTest(unittest.TestCase):
+
+    def test_something(self):
+        if not external_resource_available():
+            return  # Noncompliant {{Skip this test explicitly.}}
+        else:
+            "test".capitalize()
+            self.assertEqual(foo(), 42)
+
+    def test_something(self):
+        if not external_resource_available():
+            return  # Noncompliant {{Skip this test explicitly.}}
+#           ^^^^^^
+        else:
+            self.assertEqual
+
+    def test_with_skip(self):
+        if not external_resource_available():
+            self.skipTest("prerequisite not met")
+        else:
+            self.assertEqual(foo(), 42)
+
+    def not_a_test(self):
+        if not external_resource_available():
+            return  # OK, method is not "testXXX"
+        else:
+            self.assertEqual(foo(), 42)  # Assertion doesn't matter as the method is not discoverable by the test runner
+            print("hello")
+
+    def test_no_assertions(self):
+        if not external_resource_available():
+            return  # OK, no assertions in the test (avoid FPs)
+        else:
+            foo()
+
+class MyTestReg(unittest.TestCase):
+    def test_image_dataset_from_directory_errors(self):
+        if PIL is None:
+          return  # Noncompliant {{Skip this test explicitly.}}
+#         ^^^^^^
+
+        directory = self._prepare_directory(num_classes=3, count=5)
+
+        with self.assertRaisesRegexp(ValueError, '`labels` argument should be'):
+          _ = image_dataset.image_dataset_from_directory(
+              directory, labels=None)
+
+# Other testing library
+class MyFakeTest(faketest.TestCase):
+    # With an assert method which name exist in unittest
+    def test_assert_existing_in_unittest(self):
+        if not external_resource_available():
+            return
+        else:
+            self.assertEqual(foo(), 42)
+
+    # With an assert method which name does not exist in unittest
+    def test_assert_not_existing_in_unittest(self):
+        if not external_resource_available():
+            return
+        else:
+            self.assertFakeEqual(foo(), 42)
+
+# Multiple inheritance with specified class name
+class TestRemoveDirectory(CommandTestMixin, unittest.TestCase):
+    def test_simple_exception(self):
+        if runtime.platformType == "posix":
+            return # Noncompliant {{Skip this test explicitly.}}
+#           ^^^^^^
+        def fail(dir):
+            raise RuntimeError("oh noes")
+        self.patch(utils, 'rmdirRecursive', fail)
+        self.make_command(fs.RemoveDirectory, dict(
+            dir='workdir',
+        ), True)
+        d = self.run_command()
+
+        def check(_):
+            self.assertIn({'rc': -1}, self.get_updates(), self.builder.show())
+        d.addCallback(check)
+        return d
```

Comment:
I guess this part could be simplified with just a simple assertion (this seems copied from actual code, which removes a bit of clarity from the test).
Perhaps if you want a test where the assertion is actually called in a nested function, this could be done in a separate test.

Reply:
Good point, i cleaned the test files